gradle/javaToolchain.gradle

```diff
@@ -31,7 +31,12 @@ allprojects {
         }
     }
 
-    if (project in javaMainProjects) {
+    if (project == project(":bson-record-codec")) {
```

Comment:
This could probably be cleaner.  And I don't know how to make sure the tests only run on Java 17+

Reply:
@rozza probably will need your assistance on this one.

Reply:
No problems will take a look - this is another good reason to fix our gradle build and make it more configurable per project and explicit rather than this older approach of an uber gradle configuration.